kms-message/test/test_kms_request.c

```diff
@@ -1025,6 +1025,93 @@ kms_request_validate_test (void)
    kms_request_destroy (request);
 }
 
+/* Test private key signing.
+ *
+ * private_key_b64 was generated by taking the base64 data out of
+ * test-private-key.pem generated as follows: openssl genpkey -out
+ * test-private-key.pem -algorithm RSA -pkeyopt rsa_keygen_bits:2048
+ *
+ * It was used to generate a test signature of the string "data to sign" with:
+ * openssl dgst -sign test-private-key.pem -sha256 signme.txt | base64
+ */
+static void
+kms_signature_test (void)
+{
+   const char *private_key_b64 =
```

Comment:
In addition to this example key I generated, I tested signing a JWT I created with my own GCP credentials.